internal/database/release.go

```diff
@@ -126,7 +118,7 @@ func (repo *ReleaseRepo) Find(ctx context.Context, params domain.ReleaseQueryPar
 
 func (repo *ReleaseRepo) findReleases(ctx context.Context, tx *Tx, params domain.ReleaseQueryParams) ([]*domain.Release, int64, int64, error) {
 	queryBuilder := repo.db.squirrel.
-		Select("r.id", "r.filter_status", "r.rejections", "r.indexer", "r.filter", "r.protocol", "r.title", "r.torrent_name", "r.size", "r.timestamp", "(SELECT COUNT(*) FROM release) AS total_count").
+		Select("r.id", "r.filter_status", "r.rejections", "r.indexer", "r.filter", "r.protocol", "r.title", "r.torrent_name", "r.size", "r.timestamp", "ras.id", "ras.status", "ras.action", "ras.type", "ras.client", "ras.filter", "ras.rejections", "ras.timestamp", "(SELECT COUNT(*) FROM release) AS total_count").
 		From("release r").
 		OrderBy("r.id DESC")
```

Comment:
Found an issue with this 😄 since it fetches the actions and some releases have multiple actions, the limit becomes the total limit. So if we have `LIMIT 10` and two releases have two actions, we'll only get 8 actual releases.

Well described in this article along with some alternative fixes. Will try one of them!

Reply:
There's inherently less operations with yours (by a far margin because there's no copies), time seems to be consistent with both.
```
	subQueryBuilder := repo.db.squirrel.
		Select("*").
		From("release").
		OrderBy("id DESC")

	if params.Limit > 0 {
		subQueryBuilder = subQueryBuilder.Limit(params.Limit)
	} else {
		subQueryBuilder = subQueryBuilder.Limit(20)
	}

	if params.Offset > 0 {
		subQueryBuilder = subQueryBuilder.Offset(params.Offset)
	}

	if params.Cursor > 0 {
		subQueryBuilder = subQueryBuilder.Where(sq.Lt{"r.id": params.Cursor})
	}

	queryBuilder := repo.db.squirrel.
		Select("r.id", "r.filter_status", "r.rejections", "r.indexer", "r.filter", "r.protocol", "r.title", "r.torrent_name", "r.size", "r.timestamp",
		"ras.id", "ras.status", "ras.action", "ras.type", "ras.client", "ras.filter", "ras.rejections", "ras.timestamp",
		"(SELECT COUNT(*) FROM release) AS total_count").
		FromSelect(subQueryBuilder, "r")
```